src/core/server/core_app/core_app.ts

```diff
@@ -28,12 +29,64 @@ export class CoreApp {
     this.env = core.env;
   }
 
+  preboot(corePreboot: InternalCorePreboot, uiPlugins: UiPlugins) {
+    this.logger.debug('Prebooting core app.');
+
+    // We register app-serving routes only if there are `preboot` plugins that may need them.
+    if (uiPlugins.public.size > 0) {
```

Comment:
__note:__ This `if` isn't strictly necessary and we can get rid of it completely once we implement the `userSetup` preboot plugin (it shouldn't be possible to disable it). I just wanted to avoid the potential risk of conflicting with `503 Kibana is not ready yet` route in the meantime, but I plan that `userSetup` (or other `preboot` plugin) will replace this `503` route with the dedicated client-side app.